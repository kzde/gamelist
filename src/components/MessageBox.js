import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { colorPalettes } from '../utils/styleUtils';

const Wrapper = styled.div`
  background: ${colorPalettes.white};
  position: absolute;
  top: 50%;
  margin-left: auto;
  margin-right: auto;
  left: 0;
  right: 0;
  width: 50%;
  border-radius: 8px;
  opacity: 0.6;
`;
const Content = styled.p`
  text-align: center;
  padding: 10px 0;
`;

const MessageBox = ({ content }) => (
  <Wrapper>
    <Content>{content}</Content>
  </Wrapper>
);

MessageBox.propTypes = {
  content: PropTypes.string.isRequired
};
export default MessageBox;
