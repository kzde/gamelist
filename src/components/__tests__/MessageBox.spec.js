import React from 'react';
import renderer from 'react-test-renderer';
import MessageBox from '../MessageBox';

describe('MessageBox', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(<MessageBox children={<div>MessageBoxContent</div>} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})