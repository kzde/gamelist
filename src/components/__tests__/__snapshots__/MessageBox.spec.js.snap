// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox should renders correctly 1`] = `
<div
  className="MessageBox__Wrapper-sc-13sbgul-0 eGEFDx"
>
  <p
    className="MessageBox__Content-sc-13sbgul-1 feAuva"
  />
</div>
`;
